#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.47467241724692044 0.7153447712426013 0.5128040118415962 4.3444181243865225
  position -0.9407476307728109 2.4180369986732817 0.23951174760891175
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF Floor Floor {
  rotation 0 1 0 0.7853996938995746
  size 4 4
  tileSize 0.4 0.4
}
DEF Walls Group {
  children [
    DEF Wall Solid {
      translation -0.9898 0.075 0.4242
      rotation 0 1 0 -0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(9)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.148347 0.075 -0.417253
      rotation 0 1 0 -0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(2)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.9898 0.075 -0.4242
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(8)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.148348 0.075 0.417253
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(3)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.841349 0.075 -0.275749
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(13)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.282739 0.075 0.282862
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.84842 0.075 0.28282
      rotation 0 1 0 2.3561996938995744
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(12)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.289811 0.075 -0.275792
      rotation 0 1 0 2.3561996938995744
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(11)"
      boundingObject USE Wall
    }
  ]
}
DEF ROBOT_RED Robot {
  translation 0.165081 -0.00980597 -0.0174491
  rotation 0.014902298662239338 -0.9998849102415855 0.002843899744706686 2.38528
  children [
    Display {
      width 256
      height 256
    }
    Gyro {
    }
    Accelerometer {
    }
    Lidar {
      translation 0 0.09 0
      horizontalResolution 256
      fieldOfView 3.14159
      numberOfLayers 1
    }
    DEF MAIN_BODY Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.196078 0.196078 0.196078
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DEF LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 922.4554790595527
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
      ]
      endPoint DEF LEFT_WHEEL Solid {
        translation -0.030000000000000006 0.025 0
        rotation -0.9989441805298487 0.02750676566050536 0.03679540770866585 6.12506697864787
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.196078 0.196078
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject DEF WHEEL Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.025
              }
            }
          ]
        }
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 144.32756075465952
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint DEF RIGHT_WHEEL Solid {
        translation 0.029999999999999763 0.025 0
        rotation -1 0 0 1.5101339967086103
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.196078 0.196078
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF DS0 DistanceSensor {
      translation 0.045 0.063 0
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS1 DistanceSensor {
      translation 0.0225 0.063 -0.039
      rotation 0 1 0 1.0472003061004251
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds1"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS4 DistanceSensor {
      translation 0.039 0.063 -0.0233
      rotation 0 -1 0 -0.5236053071795865
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds4"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS2 DistanceSensor {
      translation -0.0225 0.063 -0.039
      rotation 0 1 0 2.094399693899575
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds2"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS3 DistanceSensor {
      translation -0.045 0.063 0
      rotation 0 1 0 3.141592653589793
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds3"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS5 DistanceSensor {
      translation -0.039 0.063 -0.022
      rotation 0 -1 0 -2.6179953071795863
      children [
        DEF INFRARED Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds5"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF CAMERA Camera {
      translation 0 0.045 -0.045
      children [
        Transform {
          rotation 1 0 0 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.007
              }
            }
          ]
        }
      ]
      fieldOfView 1.047
      width 1280
      height 720
      antiAliasing TRUE
      noise 0.01
    }
    DEF LED LED {
      translation 0 0.08 0
      children [
        Group {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry DEF LED_SHAPE Sphere {
                radius 0.01
              }
            }
            PointLight {
              attenuation 0 0 1
              color 0 0 0
              intensity 0.5
              location 0 0.03 0
              on FALSE
              castShadows TRUE
            }
          ]
        }
      ]
      color [
        1 0 0
        0 1 0
        0 0 1
        1 1 0
        0 1 1
        1 0 1
        1 0.647059 0
        0.627451 0.12549 0.941176
        0.545098 0.411765 0.0784314
      ]
    }
  ]
  name "ROBOT_RED"
  boundingObject DEF MYBOT_BOUNDING Transform {
    translation 0 0.0415 0
    children [
      DEF BODY Cylinder {
        height 0.08
        radius 0.045
      }
    ]
  }
  physics DEF MYBOT_PHYSICS Physics {
    density -1
    mass 0.5
  }
  controller "NMP_Controller"
}
